apps/music/js/endpoint.js

```diff
@@ -497,4 +497,24 @@ function open(blob) {
   });
 }
 
+function throttle(fn, ms) {
+  var last = Date.now();
+  var timeout;
+  return (...args) => {
+    var now = Date.now();
+    if (now < last + ms) {
+      clearTimeout(timeout);
+      timeout = setTimeout(() => {
+        last = now;
+        fn.apply(this, args);
+      }, ms);
+    }
+
+    else {
+      last = now;
+      fn.apply(this, args);
+    }
+  };
+}
```

Comment:
I see you duplicated this function above. Would it make sense to add a utils.js file or something?